### PR TITLE
upgrade tf version to 2.4

### PR DESCRIPTION
**Pull request by arthurPignet**

### CODE REVIEW

**Review by natct10 — APPROVED** [review]
Thank you @arthurPignet!!

**Review by natct10 — CHANGES_REQUESTED** [review]
I am getting dependency issues on Linux, Python3.8:
```sh
ERROR: Requested cffi>=1.0 from https://files.pythonhosted.org/packages/ff/6d/9f5004f4619e64e25785029d3fab9807eddc780e194ba323ecd4007fa8c5/cffi-1.2.0-1.tar.gz#sha256=9b9c2b964fd88ddbf99ea8fff65c56264c8866b77050fd53f0159e53c98c27e0 (from soundfile>=0.9.0->librosa==0.8.0->mplc==0.2) has different version in metadata: '1.2.0'
```

**Comment by natct10** [comment]
> I am getting dependency issues on Linux, Python3.8:
> 
> ```shell
> ERROR: Requested cffi>=1.0 from https://files.pythonhosted.org/packages/ff/6d/9f5004f4619e64e25785029d3fab9807eddc780e194ba323ecd4007fa8c5/cffi-1.2.0-1.tar.gz#sha256=9b9c2b964fd88ddbf99ea8fff65c56264c8866b77050fd53f0159e53c98c27e0 (from soundfile>=0.9.0->librosa==0.8.0->mplc==0.2) has different version in metadata: '1.2.0'
> ```

Fixed in a fresh new venv!

File: mplc/partner.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 """
 
 import numpy as np
-from keras.utils import to_categorical
 from loguru import logger
+from tensorflow.keras.utils import to_categorical
```
Comment (natct10) [inline mplc/partner.py:8]: Top!

**Review by natct10 — APPROVED** [review]
Well done!

File: mplc/mpl_utils.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def aggregate_model_weights(self):
             avg_weights_for_layer = np.average(
                 np.array(weights_for_layer), axis=0, weights=self.aggregation_weights
             )
-            new_weights.append(list(avg_weights_for_layer))
+            new_weights.append(avg_weights_for_layer)
```
Comment (bowni) [inline mplc/mpl_utils.py:135]: Is that really an innocent change?

**Review by bowni — APPROVED** [review]

File: mplc/mpl_utils.py (hunk 2 above)
Comment (arthurPignet) [reply mplc/mpl_utils.py:135]: Tf wants a list of np.array now, so we need it

**Pull request merged**